File: Java/Platform/Communication/TCPServerTest/Program1.java
import java.io.*;
import java.net.*;

class Program {

	public static void main(String[] args) throws Exception {
		//Step 1
		var listener = new ServerSocket(4010);
		service(listener);
	}

	private static void service(ServerSocket listener) throws IOException {
		var shop = Shop.open("store.xml");
		for(;;) {
			//Step 2
			var client = listener.accept();
			client.setSoTimeout(20000);
			//Step 3
			var input = client.getInputStream();
			var output = client.getOutputStream();
			var reader = new BufferedReader(new InputStreamReader(input));
			var writer = new PrintWriter(new OutputStreamWriter(output), true);
			try{
				writer.println("Welcome to CitiTek");
				String item = reader.readLine();
				String info = shop.getItemInfo(item);
				if(info != null)
					writer.println(info);
			}catch(Exception e){}
			writer.close();
			reader.close();
			//Step 4
			client.close();
		}
	}
}

File: dotnet/Platform/Communication/TcpServerTest/DemoApp/Program1.cs
using System.Net;
using System.Net.Sockets;

class Program
{
    static void Main()
    {
        var listener = new TcpListener(IPAddress.Any, 4010);
        listener.Start();
        Service(listener);
    }

    private static void Service(TcpListener listener)
    {
        Shop shop = Shop.Open("store.xml");
        for(;;)
        {
            TcpClient client = listener.AcceptTcpClient();
            client.ReceiveTimeout = 20000;
            Stream channel = client.GetStream();
            StreamReader reader = new StreamReader(channel);
            StreamWriter writer = new StreamWriter(channel) { AutoFlush = true };
            try
            {
                writer.WriteLine("Welcome to CitiTek");
                string item = reader.ReadLine();
                string info = shop.GetItemInfo(item);
                if(info != null)
                    writer.WriteLine(info);
            }
            catch {}
            writer.Close();
            reader.Close();
            client.Close();
        }
    }
}